tests/api/test_responses.py

```diff
@@ -356,10 +356,14 @@ def test_post_wrong_content_type(simple_app):
     )
     assert resp.status_code == 415
 
-    resp = app_client.post(
-        "/v1.0/post_wrong_content_type", data=json.dumps({"some": "data"})
-    )
-    assert resp.status_code == 415
+    # According to the spec, a content-type header is not required
```

Comment:
Can you validate my thinking here @Ruwann?

Reply:
According to [this](https://portswigger.net/kb/issues/00800500_content-type-is-not-specified) Probably the only way to do this, and to follow the RFC spec, is to either try to determine the bytes using something like the `magic` library or to try to parse the bytes using all available payload parsers. If either fails we should send a 415.

Reply:
We're actually doing some [best guessing](https://github.com/spec-first/connexion/pull/1588/files#diff-0f386dec8fb7dd9135bbf5c0a680ab61c9c9a9e07988be627c43b0aaf0e6692dR189) now, which is why we're no longer returning a 415 error here.

This test previously ensured that a request without a `Content-Type` header would return a 415. But as you mention as well, a `Content-Type` header is not required. So I think this test was incorrect.

Reply:
I agree, I don't think that just the absence of a `Content-Type` header warrants a 415 response. So the test is incorrect as far as I'm concerned.

However, to nitpick, if the endpoint specifies a certain `Content-Type` in the API contract, and if the `Content-Type` header is missing and the payload cannot be interpreted as that content type, it is still *valid* to return a 415. For example, [here](https://github.com/spec-first/connexion/pull/1588/files#diff-f3b50008022f534c8f329ff026c99b5bf9d12b27da6aaeae91ba2d32583f7c37R74) it would be possible to return a 415 in case of failure?

Reply:
Yeah, true. Didn't consider that a malformed payload is also an option.

Agreed, it was a bit of a sidetrack here as the original question/comment on the test is clear now.